client/state.cpp

```diff
@@ -326,11 +319,9 @@ std::vector<std::string> State::update(const torchcraft::fbs::EndGame* end) {
   std::vector<std::string> upd;
   preUpdate();
 
-  if (flatbuffers::IsFieldPresent(end, torchcraft::fbs::EndGame::VT_DATA)) {
-    if (this->update_frame(end->data())) {
-      upd.emplace_back("frame_string");
-      upd.emplace_back("frame");
-    }
+  // TODO There used to be a test here of whether a frame(diff) was present. Is it ever not?
```

Comment:
Yes, this can happen, although I can't tell you exactly when. In `Controller::endGame()`, the frame data is only added if there's a last frame (and I think there are situations where that's not the case).